Objects/typeobject.c

```diff
@@ -373,12 +373,23 @@ lookup_tp_mro(PyTypeObject *self)
 PyObject *
 _PyType_GetMRO(PyTypeObject *self)
 {
-    PyObject *mro;
+#ifdef Py_GIL_DISABLED
+    PyObject *mro = _Py_atomic_load_ptr_relaxed(&self->tp_mro);
+    if (mro == NULL) {
+        return NULL;
+    }
+    if (_Py_TryIncref(&self->tp_mro, mro)) {
```

Comment:
Is there still reference count contention? In other words, will we want to start using some sort of deferred reference count here in the future?

Reply:
I'm probably using "deferred reference count" too broadly here. I'm thinking of something like:

1) PyType_IsSubtype uses a borrowed reference to `self->tp_mro`
2) Modifications to the mro use QSBR to delay decref'ing the old MRO

Reply:
Well we're no longer ending up in the parking lot which was the large source of the slowdown, I imagine there is still some amount of contention but it's not enough to obviously show up in gdb.

Reply:
Looking at it under `perf` it does look like ~8% is spent in `_Py_DecRefShared` so this probably is still significant and using QSBR probably makes a lot of sense here especially given how little this will ever change.